### PR TITLE
Add typehint to helpers.

### PR DESCRIPTION
**Pull request by ADmad**
Added typhints to all public methods of `View/Helper.php` and all files under `View/Helper/*`.

I skipped the 2 helpers `NumberHelper` and `TextHelper` which are basically proxies to utility classes and the tests started throws tons of errors when adding typhints to them.

Also an interesting thing I found is `HtmlHelper::tag()` takes `null` and `false` as values for 1st argument `$name` (there are tests for it) which seems quite weird to me. I would prefer changing it to only take string.

### CODE REVIEW

File: src/View/Helper.php [hunk 1]
```diff
@@ -67,7 +68,7 @@ class Helper implements EventListenerInterface
     /**
      * The View instance this helper is attached to
      *
-     * @var \Cake\View\View
+     * @var \Cake\View\View|null
```
Comment (dereuromark) [inline src/View/Helper.php]: Cant be null, see constructor.

File: src/View/Helper.php [hunk 2]
```diff
@@ -122,7 +123,7 @@ public function __get($name)
      *
      * @return \Cake\View\View The bound view instance.
      */
-    public function getView()
+    public function getView(): ?View
```
Comment (dereuromark) [inline src/View/Helper.php]: Also cant be nullable then

File: src/View/Helper/FormHelper.php [hunk 3]
```diff
@@ -223,7 +224,7 @@ class FormHelper extends Helper
     /**
      * Locator for input widgets.
      *
-     * @var \Cake\View\Widget\WidgetLocator
+     * @var \Cake\View\Widget\WidgetLocator|null
```
Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: Also cant be null after constructor (at runtime).

File: src/View/Helper/FormHelper.php [hunk 4]
```diff
@@ -305,7 +306,7 @@ public function __construct(View $View, array $config = [])
      * @return \Cake\View\Widget\WidgetLocator Current locator instance
      * @since 3.6.0
      */
-    public function getWidgetLocator()
+    public function getWidgetLocator(): ?WidgetLocator
```
Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: Same, see also return doc block (not null here).

File: src/View/Helper/HtmlHelper.php [hunk 5]
```diff
@@ -797,7 +806,7 @@ public function tableCells($data, $oddTrOptions = null, $evenTrOptions = null, $
      *
      * @param array $line Line data to render.
      * @param bool $useCount Renders the count into the row. Default is false.
-     * @return string[]
+     * @return array
```
Comment (dereuromark) [inline src/View/Helper/HtmlHelper.php]: Why making this less precise?

**Review by dereuromark — COMMENTED** [review]
I agree, string $name is also the documented one.

**Comment by markstory** [comment]
Refs #11935

**Pull request merged**